charts/application-core/templates/deployment.yaml

```diff
@@ -127,23 +127,23 @@ spec:
       affinity:
         podAntiAffinity:
           preferredDuringSchedulingIgnoredDuringExecution:
-            - podAffinityTerm:
-                labelSelector:
-                  matchExpressions:
-                    - key: app
-                      operator: In
-                      values:
-                        - {{ include "application-core.fullname" . }}
-                topologyKey: topology.kubernetes.io/zone
-              weight: 100
-          requiredDuringSchedulingIgnoredDuringExecution:
             - labelSelector:
                 matchExpressions:
-                  - key: app
+                  - key: app.kubernetes.io/name
                     operator: In
                     values:
                       - {{ include "application-core.fullname" . }}
               topologyKey: "kubernetes.io/hostname"
+      topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: topology.kubernetes.io/zone
+          whenUnsatisfiable: ScheduleAnyway
```

Comment:
Given that we're only concerning ourselves with zone failure, and I've double checked that `topology.kubernetes.io/zone` is the right label here, this is sound. Also confirmed that the `maxSkew` here just changes the schedulers preference and won't block scheduleing since `whenUnsatisfiable: ScheduleAnyway` is set.